dex/src/main/scala/com/wavesplatform/dex/tool/connectors/AuthServiceRestConnector.scala

```diff
@@ -0,0 +1,55 @@
+package com.wavesplatform.dex.tool.connectors
+
+import java.nio.charset.StandardCharsets
+import java.util.concurrent.ThreadLocalRandom
+
+import cats.syntax.either._
+import com.wavesplatform.dex.auth.JwtUtils
+import com.wavesplatform.dex.domain.account.KeyPair
+import com.wavesplatform.dex.domain.crypto
+import com.wavesplatform.dex.tool._
+import com.wavesplatform.dex.tool.connectors.AuthServiceRestConnector.AuthCredentials
+import sttp.client._
+import sttp.model.Uri.QuerySegment
+
+import scala.util.Try
+
+case class AuthServiceRestConnector(target: String, chainId: Byte) extends RestConnector with JwtUtils {
+
+  def loginPageRequest: ErrorOr[Unit] = {
+    val uri = uri"$hostPortUri/swagger-ui.html"
+    for {
+      errorOrResponse <- Try { basicRequest.get(uri).send().body }.toEither.leftMap(ex => s"Cannot load swagger UI! ${ex.getMessage}")
+      _               <- errorOrResponse
+    } yield ()
+  }
+
+  private def mkAuthTokenRequestParams(keyPair: KeyPair): List[QuerySegment] = {
+    val jwtPayload = mkJwtSignedPayload(keyPair, networkByte = chainId)
+    List(
+      "grant_type" -> "password",
+      "username"   -> jwtPayload.publicKey.base58,
+      "password"   -> s"${jwtPayload.firstTokenExpirationInSeconds}:${jwtPayload.signature}",
+      "scope"      -> jwtPayload.scope.head,
+      "client_id"  -> jwtPayload.clientId
+    ).map { case (k, v) => QuerySegment.KeyValue(k, v) }
+  }
+
+  def getAuthCredentials(maybeSeed: Option[String]): ErrorOr[AuthCredentials] = {
+
+    val keyPair       = KeyPair(crypto secureHash (maybeSeed getOrElse s"minion${ThreadLocalRandom.current.nextInt}" getBytes StandardCharsets.UTF_8))
```

Comment:
Minor note:

1. `getAuthCredentials` is used only in `checkWsAccountUpdates`
2. We use `maybeSeed` here to get an account or generate a random account. That's all
3. `maybeSeed: Option[String]` in a function doesn't tell us much about "What happens if we pass None"

Suggest to use `seed: String` and generate a random account in `checkWsAccountUpdates`.